packages/insured-bridge-relayer/src/canonical-bridge-adapters/ArbitrumBridgeAdapter.ts

```diff
@@ -0,0 +1,121 @@
+import winston from "winston";
+import { ZERO_ADDRESS } from "@uma/common";
+
+import { providers, VoidSigner } from "ethers";
+import { Bridge, OutgoingMessageState } from "arb-ts";
+
+import { Outbox__factory } from "arb-ts/dist/lib/abi/factories/Outbox__factory";
+
+import BridgeAdapterInterface from "./BridgeAdapterInterface";
+
+import Web3 from "web3";
+import type { TransactionType } from "@uma/common";
+
+export class ArbitrumBridgeAdapter implements BridgeAdapterInterface {
+  public bridge: Bridge | undefined = undefined;
+
+  constructor(readonly logger: winston.Logger, readonly l1Web3: Web3, readonly l2Web3: Web3) {}
+
+  async initialize() {
+    const l1EthersProvider = new providers.Web3Provider(this.l1Web3.currentProvider as any);
+    const l1EthersVoidSigner = new VoidSigner(ZERO_ADDRESS, l1EthersProvider);
+    const l2EthersProvider = new providers.Web3Provider(this.l2Web3.currentProvider as any);
+    const l2EthersVoidSigner = new VoidSigner(ZERO_ADDRESS, l2EthersProvider);
+
+    this.bridge = await Bridge.init(l1EthersVoidSigner, l2EthersVoidSigner);
+
+    this.logger.debug({ at: "ArbitrumBridgeAdapter", message: "Initialized Arbitrum Bridge Adapter" });
+  }
+  async constructCrossDomainFinalizationTransaction(
+    l2TransactionHash: string
+  ): Promise<{ l2TransactionHash: string; finalizationTransaction: TransactionType | null }> {
+    this._throwIfNotInitialized();
+
+    if (this.bridge) {
```

Comment:
Is this just a typescript thing? I assume we know this will always be non-undefined? If so, maybe we should make this an assert rather than an if? The only reason I suggest it is that the way it reads, you need to consider the case where this condition is false (which I think should never happen).

Reply:
ye, ts was not happy seeing it can be null. I changed this to throw if null, which should never be the case, assuming you've initiated the adapter.